src/test/java/com/configcat/ConfigV2EvaluationTest.java

```diff
@@ -98,6 +102,41 @@ private static Stream<Arguments> testDataForConfigSaltAndSegmentsOverrideTest()
         );
     }
 
+    private static Stream<Arguments> testDataComparisonAttributeConversionToCanonicalStringRepresentationTest() {
+        return Stream.of(
+                Arguments.of("numberToStringConversion", .12345, "1"),
+                Arguments.of("numberToStringConversion", 0.12345, "1"),
+                Arguments.of("numberToStringConversion", .12345d, "1"),
+                Arguments.of("numberToStringConversion", 0.12345d, "1"),
+                Arguments.of("numberToStringConversion", .12345f, "1"),
+                Arguments.of("numberToStringConversion", 0.12345f, "1"),
```

Comment:
We don't need to test both `.12345` and `0.12345` because they are the same values. Also, `.12345 == .12345d` (see https://www.baeldung.com/java-literals#2-floating-point-literals)

```suggestion
                Arguments.of("numberToStringConversion", .12345, "1"),
                Arguments.of("numberToStringConversion", .12345f, "1"),
```